app/Http/Controllers/Accessories/AccessoriesController.php

```diff
@@ -127,43 +127,43 @@ public function edit($accessoryId = null)
      */
     public function update(ImageUploadRequest $request, $accessoryId = null)
     {
-        if (is_null($accessory = Accessory::find($accessoryId))) {
-            return redirect()->route('accessories.index')->with('error', trans('admin/accessories/message.does_not_exist'));
-        }
-
-        $min = $accessory->numCheckedOut();
-        $validator = Validator::make($request->all(), [
-            "qty" => "required|numeric|min:$min"
-        ]);
+        if ($accessory = Accessory::withCount('users as users_count')->find($accessoryId)) {
 
-        if ($validator->fails()) {
-            return redirect()->back()
-                ->withErrors($validator)
-                ->withInput();
-        }
-
-        $this->authorize($accessory);
+            $validator = Validator::make($request->all(), [
+                "qty" => "required|numeric|min:$accessory->users_count"
```

Comment:
This is a nice cleanup, for sure. No need to carry around methods we don't need that we use in only one place.

Reply:
I just hoisted this up a little higher (the auth gate) - because IMHO we shouldn't even be trying to validate an accessory that the user isn't allowed to be editing in the first place. Would love another set of eyes tomorrow tho.